context/ethereumContext.tsx

```diff
@@ -430,17 +443,40 @@ export const EthereumProvider: React.FC<{}> = ({ children }) => {
   const _loadOpcodes = () => {
     const opcodes: IReferenceItem[] = []
 
-    vm.getActiveOpcodes().forEach((op: Opcode) => {
+    evm.getActiveOpcodes().forEach((op: Opcode) => {
       const meta = OpcodesMeta as IReferenceItemMetaList
-      const opcode = {
-        ...meta[toHex(op.code)],
-        ...{
-          opcodeOrAddress: toHex(op.code),
-          staticFee: op.fee,
-          minimumFee: 0,
-          name: op.fullName,
-        },
-      }
+      // TODO: need to implement proper selection of doc according to selected fork (maybe similar to dynamic gas fee)
+      // Hack for "difficulty" -> "prevrandao" replacement for "merge" HF
+      const opcode =
+        selectedFork?.name === 'merge'
+          ? toHex(op.code) == '44'
+            ? {
+                ...meta['44_merge'],
+                ...{
+                  opcodeOrAddress: toHex(op.code),
+                  staticFee: op.fee,
+                  minimumFee: 0,
+                  name: 'PREVRANDAO',
+                },
+              }
+            : {
+                ...meta[toHex(op.code)],
+                ...{
+                  opcodeOrAddress: toHex(op.code),
+                  staticFee: op.fee,
+                  minimumFee: 0,
+                  name: op.fullName,
+                },
+              }
+          : {
+              ...meta[toHex(op.code)],
+              ...{
+                opcodeOrAddress: toHex(op.code),
+                staticFee: op.fee,
+                minimumFee: 0,
+                name: op.fullName,
+              },
+            }
```

Comment:
think we'd benefit from extracting selecting a doc into its own func, this piece of code is quite hard to follow otherwise.

Reply:
refactored this block of code a little bit and extracted it to a separate function.